basyx/aas/adapter/xml/xml_serialization.py

```diff
@@ -277,9 +277,11 @@ def extension_to_xml(obj: model.Extension, tag: str = NS_AAS+"extension") -> etr
                                               text=model.datatypes.XSD_TYPE_NAMES[obj.value_type]))
     if obj.value:
         et_extension.append(_value_to_xml(obj.value, obj.value_type))  # type: ignore # (value_type could be None)
-    for refers_to in obj.refers_to:
-        et_extension.append(reference_to_xml(refers_to, NS_AAS+"refersTo"))
-
+    if obj.refers_to:
```

Comment:
`Extension.refers_to` is currently declared as `Iterable[Reference]`. The problem with that is that it doesn't have to be a `List` or a `Set` for which we can checker whether they are empty or not. Thus, the iterator may be truthy, but raise a `StopIteration` immediately, resulting in an empty `refersTo` element, which isn't compliant to the schema.
We should change the declaration to something like `List` or `Set`, that support checking for emptiness.

Reply:
Fixed in 58f4597